core/model/licensing.py

```diff
@@ -2061,3 +2063,130 @@ def rights_uri_from_string(cls, rights):
+    _prioritized_content_types: Mapping[str, int]
+    _hidden_content_types: List[str]
+
+    def __init__(
+        self,
+        prioritized_drm_schemes: List[str],
+        prioritized_content_types: List[str],
+        hidden_content_types: List[str],
+    ):
+        """
+        :param prioritized_drm_schemes: The set of DRM schemes to prioritize; items earlier in the list are higher priority.
+        :param prioritized_content_types: The set of content types to prioritize; items earlier in the list are higher priority.
+        :param hidden_content_types: The set of content types to remove entirely
+        """
+
+        # Assign priorities to each content type and DRM scheme based on their position
+        # in the given lists. Higher priorities are assigned to items that appear earlier.
+        self._prioritized_content_types = {}
+        _priority = 1
+        for content_type in reversed(prioritized_content_types):
+            self._prioritized_content_types[content_type] = _priority
+            _priority = _priority + 1
+
+        self._prioritized_drm_schemes = {}
+        _priority = 1
+        for drm_scheme in reversed(prioritized_drm_schemes):
+            self._prioritized_drm_schemes[drm_scheme] = _priority
+            _priority = _priority + 1
+
+        self._hidden_content_types = hidden_content_types
+
+    def prioritize_for_pool(
+        self, pool: LicensePool
+    ) -> List[LicensePoolDeliveryMechanism]:
+        """
+        Filter and prioritize the delivery mechanisms in the given pool.
+        :param pool: The license pool
+        :return: A list of suitable delivery mechanisms in priority order, highest priority first
+        """
+        return self.prioritize_mechanisms(pool.delivery_mechanisms)
+
+    def prioritize_mechanisms(
+        self, mechanisms: List[LicensePoolDeliveryMechanism]
+    ) -> List[LicensePoolDeliveryMechanism]:
+        """
+        Filter and prioritize the delivery mechanisms in the given pool.
+        :param mechanisms: The list of delivery mechanisms
+        :return: A list of suitable delivery mechanisms in priority order, highest priority first
+        """
+
+        # First, filter out all hidden content types.
+        mechanisms_filtered: List[LicensePoolDeliveryMechanism] = []
+        for delivery in mechanisms:
+            delivery_mechanism = delivery.delivery_mechanism
+            if delivery_mechanism:
+                if delivery_mechanism.content_type not in self._hidden_content_types:
+                    mechanisms_filtered.append(delivery)
+
+        # If there are any prioritized DRM schemes or content types, then
+        # sort the list of mechanisms accordingly.
+        if (
+            len(self._prioritized_drm_schemes) != 0
+            or len(self._prioritized_content_types) != 0
+        ):
+            mechanisms_filtered.sort(
+                key=cmp_to_key(self._compare_delivery_mechanisms), reverse=True
```

Comment:
I used the `_compare_delivery_mechanisms` method in order to try to make it clearer what the policy was for a given item. Unless I'm misreading... Wouldn't the above lines reverse the list and then reverse it again?

Reply:
IMO the reverse parameter is kind of poorly named. Its really if the sort is done in ascending or descending order. So this is sorting by `content_type` in descending order, then `drm_sceme` in descending order. 

This is largely the approach that is suggested for "complex sorts" in the [python documentation](https://docs.python.org/3/howto/sorting.html#sort-stability-and-complex-sorts). See the second example there.

The [`cmp_to_key` function](https://docs.python.org/3/library/functools.html#functools.cmp_to_key), seems like its mostly a compatibility shim for older Python 2 code. I'd like to avoid using it if we can.